mgmtfn/mesosplugin/netDriver.go

```diff
@@ -0,0 +1,323 @@
+		httpError(w, "failed to set up interfaces", err)
+	}
+
+	//log.Infof("net %+v", net)
+
+	epResponse := api.VirtualizerResponse{}
+
+	// Add the service information using Service plugin
+	/*		if serviceName != "" {
+				log.Infof("Calling AddService with: ID: %s, Name: %s, Network: %s, Tenant: %s, IP: %s", cereq.EndpointID[len(cereq.EndpointID)-12:], serviceName, netName, tenantName, ep.IPAddress)
+				dnsBridge.AddService(cereq.EndpointID[len(cereq.EndpointID)-12:], serviceName, netName, tenantName, ep.IPAddress)
+			}
+	*/
+
+	// log.Infof("Sending CreateEndpointResponse: {%+v}, IP Addr: %v", epResponse, ep.IPAddress)
+	content, err = json.Marshal(epResponse)
+	if err != nil {
+		httpError(w, "Could not generate create endpoint response", err)
+		return
+	}
+
+	w.Write(content)
+}
+
+func leave(w http.ResponseWriter, r *http.Request) {
+	var (
+		content []byte
+		err     error
+		decoder = json.NewDecoder(r.Body)
+		lr      = api.VirtualizerRequest{}
+	)
+
+	logEvent("leave")
+
+	err = decoder.Decode(&lr)
+	if err != nil {
+		httpError(w, "Could not read and parse the leave request", err)
+		return
+	}
+
+	log.Infof("LeaveRequest: %+v", lr)
+
+	allocatedEndpoints.Lock()
+	epState := allocatedEndpoints.epStates[lr.ContainerID]
+
+	netID := fmt.Sprintf("%v.%v", epState.endpointRequest.NetworkName, epState.endpointRequest.TenantName)
+	endpointID := fmt.Sprintf("%v-%v", netID, lr.ContainerID)
+
+	err1 := netPlugin.DeleteEndpoint(endpointID)
+	if err1 != nil {
+		log.Infof("encountered error while deleting the endpoint: %v", err1)
+	}
+
+	// now delete from master
+	delReq := master.DeleteEndpointRequest{
+		TenantName:  epState.endpointRequest.TenantName,
+		NetworkName: epState.endpointRequest.NetworkName,
+		ServiceName: epState.endpointRequest.ServiceName,
+		EndpointID:  lr.ContainerID,
+	}
+
+	var delResp master.DeleteEndpointResponse
+	err2 := cluster.MasterPostReq("/plugin/deleteEndpoint", &delReq, &delResp)
+	if err2 != nil {
+		log.Infof("encountered error while deleting the endpoint from the master: %v", err2)
+	}
+
+	link, err := netlink.LinkByName(epState.vethMasterName)
+	if err != nil {
+		log.Infof("encountered error while trying to find eth0: %v", err)
+	} else {
+		err = netlink.LinkDel(link)
+		if err != nil {
+			log.Infof("failed to delete link: %v", err)
+		}
+	}
+	ipAddrValue, err := netutils.GetIPNumber("10.112.95.0", 24, 32, epState.vethIP)
+	if err == nil {
+		allocatedEndpoints.ipMap.Clear(ipAddrValue)
+	} else {
+		log.Infof("failed to deallocated epState.vethIP")
+	}
+
+	delete(allocatedEndpoints.epStates, lr.ContainerID)
+	allocatedEndpoints.Unlock()
+	// Send response
+	leaveResp := api.VirtualizerResponse{}
+
+	log.Infof("Sending LeaveResponse: {%+v}", leaveResp)
+
+	content, err = json.Marshal(leaveResp)
+	if err != nil {
+		httpError(w, "Could not generate leave response", err)
+		return
+	}
+
+	w.Write(content)
+}
+
+func netdGetEndpoint(epID string) (*drivers.OvsOperEndpointState, error) {
```

Comment:
Can we move this to a common package since both dockplugin and mesosplugin need it?